.github/scripts/markdown-link-check-with-retry.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -e
+
+for file in "$@"; do
+  for i in 1 2 3; do
+    if markdown-link-check --config "$(dirname "$0")/markdown-link-check-config.json" \
+                           "$file"; then
+      break
+    elif [[ $i == 3 ]]; then
+      exit 1
+    fi
+    sleep 5
+  done
+done
```

Comment:
Nit: WDYT about extracting the retry count to a variable?
```suggestion
retry_count=3

for file in "$@"; do
  for i in $(seq 1 $retry_count); do
    if markdown-link-check --config "$(dirname "$0")/markdown-link-check-config.json" \
                           "$file"; then
      break
    elif [[ $i == $retry_count ]]; then
      exit 1
    fi
    sleep 5
  done
done
```